packages/flutter_tools/bin/podhelper.rb

```diff
@@ -31,7 +31,19 @@ def flutter_additional_ios_build_settings(target)
 
   # [target.deployment_target] is a [String] formatted as "8.0".
   inherit_deployment_target = target.deployment_target[/\d+/].to_i < 9
+
+  # This podhelper script is at $FLUTTER_ROOT/packages/flutter_tools/bin.
+  # Add search paths from $FLUTTER_ROOT/bin/cache/artifacts/engine.
+  artifacts_dir = File.join('..', '..', '..', '..', 'bin', 'cache', 'artifacts', 'engine')
+  debug_framework_dir = File.expand_path(File.join(artifacts_dir, 'ios'), __FILE__)
+  release_framework_dir = File.expand_path(File.join(artifacts_dir, 'ios-release'), __FILE__)
+
   target.build_configurations.each do |build_configuration|
+    # Profile can't be derived from the CocoaPods build configuration. Use release framework (for linking only).
+    configuration_engine_dir = build_configuration.type == :debug ? debug_framework_dir : release_framework_dir
```

Comment:
This mapping is pulled from the Podfile, so flavor build configurations are handled instead of trying to parse `Flavor-Release`==`ios-release` etc:
https://github.com/flutter/flutter/blob/319a61fd817e43684cc708896975d7fad2b2d9d8/packages/flutter_tools/templates/cocoapods/Podfile-ios-swift#L7-L10